### PR TITLE
Fix: Allow scoped packages in configuration extends (fixes #2544)

### LINKED ISSUE #2544: Feature request: sharable config as a scoped package
For example, when I (`@shinnn`) publish a sharable config `eslint-config-myconfig` as a [scoped package](https://docs.npmjs.com/misc/scope), npm will place it on `node_modules/@shinnn/eslint-config-myconfig` after installing.

So, if I want to use the config via `extends` option, I should write `.eslintrc` like below:

```json
{
  "extends": "./node_modules/@shinnn/eslint-config-myconfig/index.js"
}
```

I'd like to use a scoped sharable config more simply, just like:

```json
{
  "extends": "@shinnn/myconfig"
}
```

### PR DESCRIPTION
**Pull request by eric-isakson**
Added test cases for scoped packages then fixed lib/config.js to handle them properly.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -8,3 +8,4 @@ tmp/
 .idea
 jsdoc/
 versions.json
+*.iml
```
Comment (ilyavolodin) [inline .gitignore]: What's an iml? Don't think this was intentional, right?

File: lib/config.js [hunk 2]
```diff
@@ -47,13 +47,13 @@ debug = debug("eslint:config");
...
  * conventions as require(), meaning that the first character must be nonalphanumeric
- * to be considered a file path.
+ * and not the @ sign which is used for scoped packages to be considered a file path.
  * @param {string} filePath The string to check.
  * @returns {boolean} True if it's a filepath, false if not.
  * @private
  */
 function isFilePath(filePath) {
-    return isAbsolutePath(filePath) || !/\w/.test(filePath[0]);
+    return isAbsolutePath(filePath) || !/\w|@/.test(filePath[0]);
```
Comment (ilyavolodin) [inline lib/config.js]: shouldn't this be `!/\w|\w@\w/`?

Comment (nzakas) [inline lib/config.js]: This is correct, if the first character is `@` then it's a scoped module.

File: .gitignore (hunk 1 above)
Comment (nzakas) [inline .gitignore]: It's an IntelliJ file, OK to add.

**Comment by nzakas** [comment]
Lgtm. Thanks!

**Pull request merged**